### ISSUE TITLE
Craft 4

### ISSUE THREAD

**Issue by MoritzLost** [craftcms/contact-form#215]
### Description

The plugin currently doesn't support Craft 4:

```shell
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires craftcms/contact-form 2.3.0 -> satisfiable by craftcms/contact-form[2.3.0].
    - craftcms/contact-form 2.3.0 requires craftcms/cms ^3.4.0 -> found craftcms/cms[3.4.0-beta.1, ..., 3.7.37] but it conflicts with your root composer.json require (^4.0.0-beta).
```

Is an update for Craft 4 planned? If so, is there an ETA?

### Steps to reproduce

1. Try to upgrade to the Craft 4 beta [as described here](https://craftcms.com/docs/4.x/upgrade.html#preparing-for-the-upgrade).

### Additional info

- Craft version: 4.0.0-beta1
- PHP version: 8.1.3

**Comment by JshGrn** [comment]
I am in the same boat - want to use Craft 4 on production (happy to work around errors and other bits) however this plugin wasn't supported. I have made some changes and think it works correctly - I have tested it works on the site I am building now in Craft 4 beta 3.

I will open a PR and maybe they can introduce a craft 4 branch. Some of the changes require PHP 8 in Craft 4 and thus this will not support PHP 7.4, so perhaps that's why the changes haven't been made to this plugin yet.

I have an open PR to return error codes but that is being ignored so I think this plugin is perhaps on a back burner whilst more important things are being worked on.
